web_external/Datasets/ApplyMetadataView.js

```diff
@@ -51,23 +51,25 @@ const MetadataFileCollection = Collection.extend({
     }
 });
 
-// Model for a single metadata error
-const MetadataErrorModel = Backbone.Model.extend({
+// Model for a single metadata error or warning
+const MetadataResultModel = Backbone.Model.extend({
     description: function () {
         return this.get('description');
     }
 });
 
-// Collection of metadata errors. The list of items in the collection is
-// meaningful only when initialized() is true.
-const MetadataErrorCollection = Backbone.Collection.extend({
-    model: MetadataErrorModel,
+// Collection of metadata results, i.e. errors or warnings. The 'field' property
+// should be set to the name of the field to parse in the response from the server.
+// The list of items in the collection is meaningful only when initialized() is true.
+const MetadataResultModelCollection = Backbone.Collection.extend({
+    model: MetadataResultModel,
+    field: null,
```

Comment:
You can add an `initialize` method to the `MetadataResultModelCollection` to take `field`.

Just call `Backbone.Collection.prototype.apply(this, arguments);` as the first line, to make it transparent.

Reply:
I added an `initialize` method to accept the `field`.

Reply:
The default Backbone.Collection `initialize()` is documented as empty, so I think it's common to omit the prototype call in this case.

Reply:
Agreed.